src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionActionServiceImpl.java

```diff
@@ -225,6 +230,25 @@ public void handleCheckInCreation(CheckInDTO checkIn) {
     }
   }
 
+  private void associateNewLoanWithPatronTransaction(StorageLoanDTO loan, InnReachTransaction transaction) {
+    log.info("Associating a new loan {} with patron transaction {}", loan.getId(), transaction.getId());
+    var hold = transaction.getHold();
+    hold.setFolioLoanId(loan.getId());
+    hold.setDueDateTime(toEpochSec(loan.getDueDate()));
+  }
+
+  private void associateNewLoanWithLocalTransaction(StorageLoanDTO loan, InnReachTransaction transaction) {
+    if (transaction.getState() == LOCAL_HOLD) {
```

Comment:
Not sure if the requirements about `LOCAL_HOLD` refers to transaction state or type. Clarify with PO if there is need to check the state

> If the associated INN-Reach transaction is a LOCAL_HOLD